modules/custom/dkan_api/src/Storage/DrupalNodeDataset.php

```diff
@@ -2,87 +2,148 @@
 
 namespace Drupal\dkan_api\Storage;
 
-use Drupal\node\Entity\Node;
-use Contracts\Storage;
-use Contracts\BulkRetriever;
+use Drupal\dkan_common\Storage\StorageInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * DrupalNodeDataset.
+ */
+class DrupalNodeDataset implements StorageInterface {
+
+  /**
+   * Entity Type Manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   Injected entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * Get the node storage.
+   *
+   * @return \Drupal\node\NodeStorageInterface
+   *   Node Storage.
+   */
+  protected function getNodeStorage() {
+    return $this->entityTypeManager
+      ->getStorage('node');
+  }
 
-class DrupalNodeDataset implements Storage, BulkRetriever {
+  /**
+   * @return string
+   *   Type of node.
+   */
   protected function getType() {
     return 'data';
   }
 
+  /**
+   * {@inheritDoc}.
+   */
   public function retrieve(string $id): ?string {
 
-    foreach ($this->getNodesByUuid($id) as $result) {
-      $node = Node::load($result->nid);
+    if (FALSE !== ($node = $this->getNodeByUuid($id))) {
       return $node->field_json_metadata->value;
     }
 
     throw new \Exception("No data with the identifier {$id} was found.");
   }
 
+  /**
+   * {@inheritDoc}.
+   */
   public function retrieveAll(): array {
-    $connection = \Drupal::database();
-    $sql = "SELECT nid FROM node WHERE type = :type";
-    $query = $connection->query($sql, [':type' => $this->getType()]);
-    $results = $query->fetchAll();
+
+    $nodeStorage = $this->getNodeStorage();
+
+    $nodeIds = $nodeStorage->getQuery()
+      ->condition('type', $this->getType())
+      ->execute();
 
     $all = [];
-    foreach ($results as $result) {
-      $node = Node::load($result->nid);
+    foreach ($nodeIds as $nid) {
```

Comment:
Lets keep the Drupal core standard of making our function variables different from our class properties. So instead of $nodeIds lets use $node_ids

Reply:
I'll change it back but I thought drupal 8 had moved to PSR-0/PSR-2 standards so camel case was the norm for properties and variables. The  `drupal/coder` ruleset didn't flag it as an issue.

Reply:
fixed in 2b63599093fe9c684448bc2d4523719e5e018c9c